### PR TITLE
[FIX] CanonicalProcess#withoutDsiabledNodes

### PR DESCRIPTION
**Pull request by nadberezny**
1. on rewrite do not cut off disabled subprocess output

### CODE REVIEW

File: engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/canonicalgraph/CanonicalProcess.scala [hunk 1]
```diff
@@ -10,41 +10,53 @@ sealed trait CanonicalTreeNode
...
-            }.filterNot { case (_, canonicalNodes) => canonicalNodes.isEmpty }
+  private def withoutDisabled(nodes: List[CanonicalNode]): List[CanonicalNode] =
+    nodes.foldLeft(List.empty[CanonicalNode]) { (agg, node) =>
+      node match {
+        case flatNode: canonicalnode.FlatNode if isNodeDisabled(flatNode) =>
+          agg
+        case filterNode: canonicalnode.FilterNode if isNodeDisabled(filterNode) =>
+          agg
+        case subprocessNode: canonicalnode.Subprocess if isNodeDisabled(subprocessNode) =>
+          subprocessNode.outputs.headOption match {
```
Comment (arkadius) [inline engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/canonicalgraph/CanonicalProcess.scala]: Why head is special? What about tail?

File: engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/canonicalgraph/CanonicalProcess.scala [hunk 2]
```diff
@@ -10,41 +10,53 @@ sealed trait CanonicalTreeNode
...
-        case split: canonicalnode.SplitNode =>
-          split.copy(nexts = split.nexts.map(withoutDisabled).filterNot(_.isEmpty))
-
-        case subprocess: canonicalnode.Subprocess =>
-          subprocess.copy(
-            outputs = subprocess.outputs.map { case (id, canonicalNodes) =>
-              (id, withoutDisabled(canonicalNodes))
-            }.filterNot { case (_, canonicalNodes) => canonicalNodes.isEmpty }
+  private def withoutDisabled(nodes: List[CanonicalNode]): List[CanonicalNode] =
+    nodes.foldLeft(List.empty[CanonicalNode]) { (agg, node) =>
```
Comment (arkadius) [inline engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/canonicalgraph/CanonicalProcess.scala]: It looks like it is simple transformation 1 to 0...n, why not to just use `flatMap`?

File: engine/interpreter/src/test/scala/pl/touk/nussknacker/engine/canonicalgraph/CanonicalProcessTest.scala [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+  val disabledFilter1 = FilterNode(data = Filter("filter1", "''", isDisabled = Some(true)), List(sink1))
+
+  def subprocess(output: List[CanonicalNode], isDisabled: Boolean) =
+    Subprocess(
+      SubprocessInput(
+        "sub1",
+        SubprocessRef("sub1", Nil),
+        isDisabled = Some(isDisabled)
+      ),
+      Map("subOut" -> output)
```
Comment (arkadius) [inline engine/interpreter/src/test/scala/pl/touk/nussknacker/engine/canonicalgraph/CanonicalProcessTest.scala]: Can you provide test for subprocess with many outputs?

File: engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/canonicalgraph/CanonicalProcess.scala (hunk 1 above)
Comment (nadberezny) [reply engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/canonicalgraph/CanonicalProcess.scala]: Subprocess resolver on canonical compilation allows a subprocess to be disabled only if one output is defined.

File: engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/canonicalgraph/CanonicalProcess.scala (hunk 2 above)
Comment (nadberezny) [reply engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/canonicalgraph/CanonicalProcess.scala]: :+1:

File: engine/interpreter/src/test/scala/pl/touk/nussknacker/engine/canonicalgraph/CanonicalProcessTest.scala (hunk 3 above)
Comment (nadberezny) [reply engine/interpreter/src/test/scala/pl/touk/nussknacker/engine/canonicalgraph/CanonicalProcessTest.scala]: :+1:

**Pull request merged**